File: 17s2/lab06/echon.py
#!/usr/bin/python

import sys

if len(sys.argv) == 3:
	try:
		circulation = int(sys.argv[1])

	except ValueError:
		circulation = "ERROR"

	if isinstance(circulation,int):
		if circulation>=0:
			count = 0
			while count < circulation:
				print(sys.argv[2])
				count+=1
		else:
			print("{}: argument 1 must be a non-negative integer".format(sys.argv[0]))
			sys.exit(1)
	else:
		print("{}: argument 1 must be a non-negative integer".format(sys.argv[0]))
		sys.exit(1)

else:
	print("Usage: {} <number of lines> <string>".format(sys.argv[0]))
	sys.exit(1)

File: 17s2/lab05/echon.pl
#!/usr/bin/perl -w

if (@ARGV == 2 and $ARGV[0] =~ /^[0-9]*$/ and $ARGV[1] =~ /^.*$/ ) {
   foreach ($c=0;$c<$ARGV[0];$c++) {
        print "$ARGV[1]";
        print "\n";
   }
}
elsif ($ARGV[0] !~ /^[0-9]*$/) {
    print "./echon.pl: argument 1 must be a non-negative integer\n";
    exit 1;
}

else {
    print "Usage: ./echon.pl <number of lines> <string>\n";
    exit 1;
}
